Add schema migrator to fix upgrade problem (#3102)
cloudflare/terraform-provider-cloudflare: .changelog/3102.txt

```release-note:bug

> resource/email_routing_rule: Fix issue with upgrading resource from 4.22.0 to 4.23.0

Reviewer: ```suggestion
resource/email_routing_rule: add schema migration for upgrading 4.22.0 to 4.23.0
```
